src/main/java/com/google/api/codegen/config/GrpcGapicRetryMapping.java

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import com.google.api.codegen.RetryCodesDefinitionProto;
+import com.google.api.codegen.RetryParamsDefinitionProto;
+import com.google.api.codegen.grpc.MethodConfig.Name;
+import com.google.api.codegen.grpc.MethodConfig.RetryOrHedgingPolicyCase;
+import com.google.api.codegen.grpc.MethodConfig.RetryPolicy;
+import com.google.api.codegen.grpc.ServiceConfig;
+import com.google.api.tools.framework.model.Interface;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.util.Durations;
+import java.util.HashMap;
+import java.util.Map;
+
+@AutoValue
+public abstract class GrpcGapicRetryMapping {
+
+  public static GrpcGapicRetryMapping create(
+      ServiceConfig config, ImmutableMap<String, Interface> protoInterfaces) {
+    Map<String, String> methodCodesMap = new HashMap<>();
```

Comment:
nit: you can make `methodCodesMap` type of `ImmutableMap.Builder<String, String>`, and when finished with adding elements to it just call `methodCodesMap.build()` and you have the `ImmutableMap`. This way you don't have to make a copy.

Reply:
I started with that, but ran into the issue of duplicate entries because the `ImmutableMap.Builder` doesn't have a 1. `putIfAbsent` 2. `containsKey` 3. `get`. I could've used another map as the "reference map" but felt that would be weird. 

Open to suggestions though. WDYT?

Reply:
That makes sense.